dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -183,10 +183,7 @@ public Task<SKContext> RunAsync(string input, CancellationToken cancellationToke
     /// <inheritdoc/>
     public async Task<SKContext> RunAsync(ContextVariables variables, CancellationToken cancellationToken, params ISKFunction[] pipeline)
     {
-        var context = new SKContext(
-            variables,
-            this._skillCollection,
-            this.LoggerFactory);
+        var context = new SKContext(this, variables);
```

Comment:
If the SKContext will contain a kernel reference -- couldn't the implementations of InvokeAsync do the handling of the events instead of RunAsync?

Reply:
@lemillermicrosoft Originally that was my idea implemented in this PR #2045 but it was thrown away in favor of the Kernel because `ISKFunction` shouldn't have additional responsibilities on how to handle events.